linera-client/src/client_options.rs

```diff
@@ -517,10 +517,19 @@ pub enum ClientCommand {
         #[arg(long)]
         maximum_fuel_per_block: Option<u64>,
 
-        /// Set the maximum size of an executed block.
+        /// Set the maximum size of an executed block, in bytes.
         #[arg(long)]
         maximum_executed_block_size: Option<u64>,
 
+        /// Set the maximum size of data blobs, compressed bytecode and other binary blobs,
+        /// in bytes.
+        #[arg(long)]
+        maximum_blob_size: Option<u64>,
+
+        /// Set the maximum size of decompressed contract or service bytecode, in bytes.
+        #[arg(long)]
+        maximum_bytecode_size: Option<u64>,
```

Comment:
These are part of the `ClientOptions` struct but they're in a sense a protocol-level parameters? i.e. all clients (and validators) of the particular chain (in particular epoch configuration) have to agree on these? Is it enforced in any other way than a protocol failure if they mismatch between participants (i.e. when client uses a limit higher than a validator he might propose a block which fails during validation)? Can it be downloaded from the chain it connects to?

Reply:
They are only options of `resource-control-policy` and `create-genesis-config`, so they are a setting that affects the whole network. Clients cannot choose their values unless they are starting a whole new network or own the admin chain.